tezos/encoding-derive/src/encoding.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use parse_display::{Display, FromStr};
+use proc_macro2::Span;
+
+#[derive(Debug)]
+pub struct DataWithEncoding<'a> {
+    pub name: &'a syn::Ident,
+    pub encoding: Encoding<'a>,
+}
+
+#[derive(Debug)]
+pub struct StructEncoding<'a> {
+    pub name: &'a syn::Ident,
+    pub fields: Vec<FieldEncoding<'a>>,
+}
+
+#[derive(Debug)]
+pub struct FieldEncoding<'a> {
+    pub name: &'a syn::Ident,
+    pub encoding: Option<Encoding<'a>>,
+}
+
+#[derive(Debug)]
+pub struct EnumEncoding<'a> {
+    pub name: &'a syn::Ident,
+    pub tag_type: syn::Ident,
+    pub tags: Vec<Tag<'a>>,
+}
+
+#[derive(Debug)]
+pub struct Tag<'a> {
+    pub id: syn::LitInt,
+    pub name: &'a syn::Ident,
+    pub encoding: Encoding<'a>,
+}
+
+#[derive(Debug)]
+pub enum Encoding<'a> {
+    Unit,
+    Primitive(PrimitiveEncoding, Span),
+    Bytes(Span),
+    Path(&'a syn::Path),
+    Z(Span),
+    MuTez(Span),
```

Comment:
It might be nom-specific but what are `Z` and `MuTez` ? 
Could we find more explicit names ?

Reply:
Well, `Z` stands for Z-arith numbers, and that was used long ago in both Tezedge and Tezos code bases. `MuTez` is `micro-Tez`, again widely used term in Tezos. I agree that `Z` can be more verbose.